forge-game/src/main/java/forge/game/ability/effects/DestroyEffect.java

```diff
@@ -71,12 +73,37 @@ public void resolve(SpellAbility sa) {
             if (gameCard == null || !tgtC.equalsWithGameTimestamp(gameCard)) {
                 continue;
             }
-            internalDestroy(gameCard, sa, params, zoneMovements);
+            if (sa.hasParam("Sacrifice")) {
+                sacList.add(gameCard);
+            } else {
+                internalDestroy(gameCard, sa, params, zoneMovements);
+            }
         }
 
         for (final Card unTgtC : untargetedCards) {
             if (unTgtC.isInPlay()) {
-                internalDestroy(unTgtC, sa, params, zoneMovements);
+                if (sa.hasParam("Sacrifice")) {
+                    sacList.add(unTgtC);
+                } else {
+                    internalDestroy(unTgtC, sa, params, zoneMovements);
+                }
+            }
+        }
+
+        final boolean remDestroyed = sa.hasParam("RememberDestroyed");
+
+        SpellAbility cause = sa;
+        if (sa.isReplacementAbility()) {
+            cause = (SpellAbility) sa.getReplacingObject(AbilityKey.Cause);
+        }
+
+        // TODO remove Sacrifice inside Destroy
+        for (Card c : game.getAction().sacrifice(sacList, cause, true, params)) {
```

Comment:
@tool4ever if we would remove the Destroy/Sacrifice thing, I could drop this extra piece of code

Reply:
Waiting for #5744 to be merged first